### PR TITLE
search is no longer showing posts in draft/published state

### PR DESCRIPTION
**Pull request by babiracka**
- query is not repeated

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -27,27 +27,25 @@ def update
...
+
   def index
-    @posts = Post.all
+    @posts = Post.published.order('views_count DESC, created_at DESC')
     if params[:query].present?
-      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
+      @posts = @posts.search(params[:query])
     else
-      @posts = Post.published.order('views_count DESC , created_at DESC')
+      @posts = @posts
```
Comment (rishijain) [inline app/controllers/posts_controller.rb:47]: we do not need to add this else condition now as you have collected all the published posts earlier. So we can go away with the else condition.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -27,27 +27,25 @@ def update
...
     end
-    Statistic.increment_counter(:count, Statistic.home_page_view_count.id)
```
Comment (rishijain) [inline app/controllers/posts_controller.rb:50]: Also any particular reason to remove this ?

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (babiracka) [inline app/controllers/posts_controller.rb:47]: I'm sorry, I PRed too fast, that's why I closed it - I'm fixing the code right now.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -27,27 +27,25 @@ def update
...
-#     if query.present?
-#       Post.index
-#     else
-#       scoped
-#     end
-#   end
-
-#   def search
-#     Post.search params[:query]
-#   end
```
Comment (rishijain) [inline app/controllers/posts_controller.rb:41]: Before you start fixing any bugs or make any changes, please keep your master up to date with main master repo.

**Pull request closed without merging**